docs/testplan/PFC_Snappi_Additional_Testcases.md

```diff
@@ -0,0 +1,65 @@
+|           12 | PFCWD (MACSEC)           | **TEST PFCWD BEHAVIOR OVER MACSEC ENABLED LINK:** <br>Aim of the test case is to ensure that PFCWD drops the lossless priority traffic on detecting PFC storm on MACSEC enabled DUT and mix of lossless and lossy traffic.                                                                                                | - Single 100Gbps Ingress and single 100Gbps egress - PFCWD(Drop) over MACSEC <br>- Repeat with Single 400Gbps ingress and Single 400Gbps egress. <br> - 1024B packet-size.                                                             | - 90% - Lossless Priority 3 and 4<br>- 10% - Lossy Background Flow [Priority 0, 1, and 2] <br>3. Enabled PFCWD in default mode before the test. <br>4. Send pause to the egress interface of DUT from IXIA Rx.                                                         | - Single LC - Multiple ASIC <br>- Multiple LC                                    | 1. There should be packet drops on the DUT (IXIA Tx packets > IXIA Rx packets). <br>2. There should PFCs on DUT egress but no PFC sent from DUT ingress to traffic transmitter. <br>3. Only 10% of lossy traffic should flow. <br>4. PFCWD drop counter should increment <br>5. DUT Rx port should be in 'stormed' mode.                                                                                                                                                                       |
+|           13 | ECN                      | **APPROPRIATE ECN MARKING ON CONGESTION RESTORATION:**<br>Aim of the test case is to ensure that DUT marks the lossless priority traffic with appropriate ECN flags when congestion is detection and restoration.                                                                                                         | - Single 100Gbs ingress and single 100Gbps egress <br>- Repeat with 400Gbps ingress and egress. <br>- 1024B packet-size <br>- ECN congestion and packet-marking test.<br>                                                              | - Disable PFCWD <br>- 90% - Lossless Priority 3 and 4<br>- 10% - Lossy Background Flow [Priority 0, 1, and 2] <br>- Send PFC storm to the egress implying ECN congestion. <br>- Stop the PFC storm. <br>- Monitor set of packets to ensure packets are ECN-marked.     | - Single LC - Multiple ASIC <br>- Multiple LC                                    | 1. PFC storm should stop the packets coming DUT ingress. <br>2. No packets going out of egress port during storm. <br>3. The captured packet on egress, once storm is stopped should be ECN-marked.                                                                                                                                                                                                                                                                                            |         

```

Comment:
Another approach to test this can be to send the continous XOFF rate, but change the quanta in the XOFF.
Say, 1000, 2000, 4000, 8000, 16000, 32000, 65535 quanta units.
For each case, count the ECN marked frames received. Below a certain quanta, the queue building up will not exceed the min-threshold and thus, no marking. Then as quanta increases, the marking increases as the queue depth increases. For quanta that causes queue threshold to exceed the max threshold, all packets will be marked.

Reply:
@alpeshspatel  - I think this is very much specific to Cisco. It will be good if Cisco adds the necessary test case for the same and track it via a different PR.

Thanks,
-A